src/gui/inputdevices/TouchInputHandler.cpp

```diff
@@ -137,13 +127,18 @@ void TouchInputHandler::zoomStart() {
         zoomControl->setZoomFitMode(false);
     }
 
+    auto center = (this->priLastAbs + this->secLastAbs) / 2.0;
+
     auto* mainWindow = inputContext->getView()->getControl()->getWindow();
+    int rx, ry;
+    gdk_window_get_root_coords(*mainWindow, 0, 0, &rx, &ry);
+    center.x -= rx;
+    center.y -= ry;
```

Comment:
~~This part might warrant a comment. (I'm not very familiar with GTK, though, so it could just be that :)).~~

Consider moving this code into a method in `MainWindow.h/MainWindow.cpp`. I think this will make it easier to understand what's going on here & limit coupling between the input system and GTK.

Reply:
> I replaced `event->x` with `event->x_root` as well, so instead of calculating the offset as double it calculates the center as integers independent of the scroll value. I don't understand where `event->x` comes from and replacing it in this way seemed to fix the page jump problem when zooming with ctrl+mousewheel. But on the other hand, from googling it seemed like `event->x_root` would be in window space, but it was in screen space for me...

I just saw this! It might be worth documenting this here!